lib/graphql/client.ex

```diff
@@ -29,7 +29,7 @@ defmodule BridgeEx.Graphql.Client do
   @spec call(
           url :: String.t(),
           query :: String.t(),
-          variables :: map(),
+          variables :: map() | String.t(),
```

Comment:
I'm not sure this is the right fix. I think the solution is to encode variables right there and not before this function call so that you are sure that variables are always maps.

Reply:
Technically the fix on the typespec is right as it is in accordance with the current code. 🙂 
But I agree with you that we could do some refactor...

Reply:
+1 for fixing encoding variables, team app encountered the same typespec error today using `encode_variables: true`

Reply:
I thought about this and I see some options:

1. Include the info needed to know whether to escape the variables or not directly in the type (for instance `{:escape | :no_escape, map()}` in place of the current `map()` - or alternatively we could use a struct).
2. Add another option argument, like `escape_variables: true`. However, IMHO it currently has too many "options" arguments!
3. Completely refactor the client to use a builder approach, that is more flexible from the user perspective, for example:
```elixir
url
|> GraphQL.query(query)
|> GraphQL.variables(vars, escape: true)
|> GraphQL.header("authorization", token)
|> GraphQL.retry(max_retries: 1, timing: :exponential)
|> GraphQL.run()
```

Almost all these approaches are not so trivial, and don't are backward compatible, so I would prefer to discuss this refactor in a specific PR. Do you agree?

Reply:
I agree that we could create a new issue that details this change and then we can do it in a later PR